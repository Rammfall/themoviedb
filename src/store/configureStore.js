import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
