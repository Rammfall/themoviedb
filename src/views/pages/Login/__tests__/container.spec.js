import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import LoginPage from '../container'

describe('LoginPage', () => {
  const mockStore = configureStore()()

  it('matches snapshot', () => {
    const container = shallow(
      <LoginPage
        store={mockStore}
        onSubmit={jest.fn}
      />
    )
      .dive()
      .dive()

    expect(container).toMatchSnapshot()
  })
})
