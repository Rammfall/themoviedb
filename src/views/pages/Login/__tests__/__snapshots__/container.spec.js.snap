// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage matches snapshot 1`] = `
<Formik
  handleSubmit={[Function]}
  initialValues={
    Object {
      "password": "",
      "username": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={[Function]}
>
  <Component />
</Formik>
`;
